components/automatic-points-table/commons/automatic-points-table.lua

```diff
@@ -54,4 +58,76 @@ function AutomaticPointsTable:parseTournaments(args)
 	return tournaments
 end
 
+function AutomaticPointsTable:parseTeams(args)
+	local teams = {}
+	for _, team in Table.iter.pairsByPrefix(args, 'team') do
+		local parsedTeam = Json.parse(team)
+		parsedTeam.aliases = self:parseAliases(parsedTeam)
+		parsedTeam.deductions = self:parseDeductions(parsedTeam)
+		parsedTeam.manualPoints = self:parseManualPoints(parsedTeam)
+		table.insert(teams, parsedTeam)
+	end
+	return teams
+end
+
+--- Parses the team aliases, used in cases where a team is picked up by an org or changed name in some
+--- of the tournaments, in which case aliases are required to correctly query the team's results & points
+function AutomaticPointsTable:parseAliases(team)
+	local aliases = {}
+	for key, value in pairs(team) do
+		if type(key) == 'string' and string.find(key, 'alias%d+') then
+			local aliasIndexString = String.split(key, 'alias')[1]
+			local aliasIndex = tonumber(aliasIndexString)
+			local aliasName = value
+			aliases[aliasIndex] = aliasName
+		end
+	end
+	return aliases
+end
+
+--- Parses the teams' deductions, used in cases where a team has disbanded or made a roster change
+--- that causes them to lose a portion or all of their points that they've accumulated up until that change
+function AutomaticPointsTable:parseDeductions(team)
+	local deductions = {}
+	for key, value in pairs(team) do
+
+		if type(key) == 'string' then
+			if string.find(key, 'deduction%d+note') then
+				local deductionIndex = tonumber(string.match(key, '%d+'))
+				local deductionNote = value
+
+				if not deductions[deductionIndex] then
+					deductions[deductionIndex] = {}
+				end
+
+				deductions[deductionIndex].note = deductionNote
+			elseif string.find(key, 'deduction%d+') then
+				local deductionIndex = tonumber(String.split(key, 'deduction')[1])
+				local deductionAmount = value
+
+				if not deductions[deductionIndex] then
+					deductions[deductionIndex] = {}
+				end
+
+				deductions[deductionIndex].amount = tonumber(deductionAmount)
+			end
+		end
+	end
+	return deductions
+end
+
+function AutomaticPointsTable:parseManualPoints(team)
+	local manualPoints = {}
+	for key, value in pairs(team) do
+		if type(key) == 'string' and string.find(key, 'points%d+') then
+
+			local pointsIndex = tonumber(String.split(key, 'points')[1])
+			local points = tonumber(value)
+
+			manualPoints[pointsIndex] = points

```

Comment:
```suggestion
	for index = 1, tournamentCount do
		if String.isNotEmpty(team['points' .. index]) then
			manualPoints[index] = tonumber(team['points' .. index])
```

Reply:
FYI, you can stack these comments and avoid some of the spam

Reply:
Do I edit the original comment to achieve that?

Reply:
Instead of "submit single comment", start a review. Then you can stack all these suggestions by not selecting "commit suggestion", but "add suggestion to batch" instead.

Reply:
Alright thanks, never used some of these features before so I'm happy to learn how to better use these features